### DISCUSSION TITLE
Drill_To_Detail - Have Feature to pick and choose columns to show in Detail from the dataset

### DISCUSSION THREAD

**Discussion by maheshkumarkr** [apache/superset#24500] [Ideas]
Hello,

Current Superset version : 2.1.0
Feature : Drill_To_Detail
Request:
We are currently testing 2.1.0s features - Drill_To_Detail - when using that inside our charts/dashboard - we are able to get the Detail all good - but that Detail currently shows all columns in the dataset - for our use cases, we could have multiple columns (sometimes more than 100) in dataset, so instead of showing all columns in the detail view - is it possible to pick and choose columns before-hand to be displayed as part of detail view ?

Thank you.

**Comment by lf-floriandin** [comment]
Good idea!

Feel like there are some features missing for working with big datasets.
[This one](https://github.com/apache/superset/discussions/24351) goes into a similar direction. Would be nice to have same way to improve working with bigger datasets.

**Comment by chetansharma47** [comment]
we can work on sample Datasource of a dataset.

**Comment by maheshkumarkr** [comment]
Hello Chetan,

By sample data source , what are you referring to ? - Drill To detail
currently loads data from the dataset based on the drilled field /
condition - but it shows all columns from source dataset - so that is where
my question came to be - to be able to pick and choose columns instead of
having all columns from dataset.

Thank you.

**Reply by chetansharma47** [reply]
Hello Maheshkumarkr, 

noticed in superset, enable the flag of DRILL_TO_DETAIL , it work but when open any field that called access denied because if you are not giving access  "can sample on Datasource" you can not  access fields. if is this help related to exclude columns.

Thank you.
